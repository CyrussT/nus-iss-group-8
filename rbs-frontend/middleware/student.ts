export default defineNuxtRouteMiddleware(() => {
    const authStore = useAuthStore()

    if (authStore.user.value?.role !== 'STUDENT') {
      return navigateTo('/unauthorized')
    }
  })