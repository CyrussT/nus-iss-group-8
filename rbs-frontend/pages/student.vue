<template>
  <div>
    <p>Student Page</p>
  </div>
<!-- 
  <div v-if="emergencyMessage" class="bg-red-100 p-4 rounded-xl text-red-700 font-semibold shadow mb-4">
    ğŸš¨ {{ emergencyMessage }}
  </div> -->
</template>

<script lang="ts" setup>
definePageMeta({
  middleware: ['auth', 'student']
})

// const emergencyMessage = ref<string | null>(null);

// onMounted(() => {
//   const socket = new SockJS('http://localhost:8080/ws');
//   const stompClient = Stomp.over(socket);

//   stompClient.connect({}, () => {
//     stompClient.subscribe('/topic/emergency', (message) => {
//       emergencyMessage.value = message.body;
//       alert(`ğŸš¨ Emergency Notice: ${message.body}`);  // You can make a custom popup later
//     });
//   });
// });
</script>