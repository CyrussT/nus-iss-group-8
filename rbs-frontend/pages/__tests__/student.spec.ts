import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import student from '../student.vue';

describe('student', () => {
  it('renders correctly', () => {
    const wrapper = mount(student, {
      global: {
        stubs: {
          UButton: true,
          UIcon: true
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
