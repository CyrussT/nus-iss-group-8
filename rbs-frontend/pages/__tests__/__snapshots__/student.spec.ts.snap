// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`student > renders correctly 1`] = `
"<div>
  <p>Student Page</p>
</div>
<!-- 
  <div v-if="emergencyMessage" class="bg-red-100 p-4 rounded-xl text-red-700 font-semibold shadow mb-4">
    ğŸš¨ {{ emergencyMessage }}
  </div> -->"
`;
